tests/integration/integration_test.go

```diff
@@ -800,6 +801,37 @@ func testConstructMethodsUnknown(t *testing.T, lang string, dependencies ...stri
 	}
 }
 
+func buildTestProvider(t *testing.T, providerDir string) string {
+	fn := func() {
+		providerName := "pulumi-resource-testprovider"
+		if runtime.GOOS == "windows" {
+			providerName += ".exe"
```

Comment:
I'm curious if this is needed; in a related PR @dixler was not modifying the path  with .exe https://github.com/pulumi/pulumi/pull/10731 and it seemingly passed Windows tests?

Reply:
This is the opposite side of the same thing - we have a helper method for locating a plugin which automatically adds ".exe" to the search string.

Consequently we must make sure we add ".exe" when building. In the makefile diff we did the same thing: 

```
go build -o testprovider$(shell go env GOEXE)
```

Or equivalent.